src/codegen.cpp

```diff
@@ -1396,16 +1399,7 @@ void *jl_get_llvmf(jl_function_t *f, jl_tupletype_t *tt, bool getwrapper, bool g
         return llvmf;
     }
 
-    if (linfo->fptr && !jl_getUnwindInfo((uintptr_t)linfo->fptr)) {
-        // Not in in the current ExecutionEngine
-        // found in the system image: force a recompile
-        linfo->functionObjects.specFunctionObject = NULL;
-        linfo->functionObjects.functionObject = NULL;
-        linfo->fptr = NULL;
-    }
-    if (linfo->functionObjects.functionObject == NULL) {
-        jl_compile_linfo(linfo, NULL);
-    }
+    assert(getdeclarations);
```

Comment:
this assert is redundant with the branch above. there doesn't seem to be a path that ensures `jl_compile_linfo` has been run on linfo. did you run the `reflection.jl` tests?